pkg/controllers/nodepool/readiness/suite_test.go

```
@@ -0,0 +1,119 @@
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/awslabs/operatorpkg/status"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "knative.dev/pkg/logging/testing"
+
+	"sigs.k8s.io/karpenter/pkg/apis"
+	"sigs.k8s.io/karpenter/pkg/apis/v1beta1"
+	"sigs.k8s.io/karpenter/pkg/cloudprovider/fake"
+	nodepoolreadiness "sigs.k8s.io/karpenter/pkg/controllers/nodepool/readiness"
+	"sigs.k8s.io/karpenter/pkg/test"
+	. "sigs.k8s.io/karpenter/pkg/test/expectations"
+)
+
+var (
+	nodePoolReadinessController *nodepoolreadiness.Controller
+	ctx                         context.Context
+	env                         *test.Environment
+	cloudProvider               *fake.CloudProvider
+	nodePool                    *v1beta1.NodePool
+	nodeClass                   *v1test1.TestNodeClass
+)
+
+func TestAPIs(t *testing.T) {
+	ctx = TestContextWithLogger(t)
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Counter")
+}
+
+var _ = BeforeSuite(func() {
+	cloudProvider = fake.NewCloudProvider()
+	env = test.NewEnvironment(test.WithCRDs(apis.CRDs...), test.WithCRDs(v1test1.CRDs...))
+	nodePoolReadinessController = nodepoolreadiness.NewController(env.Client, cloudProvider)
+})
+var _ = AfterEach(func() {
+	ExpectCleanedUp(ctx, env.Client)
+})
+
+var _ = AfterSuite(func() {
+	Expect(env.Stop()).To(Succeed(), "Failed to stop environment")
+})
+
+var _ = Describe("Counter", func() {
+	BeforeEach(func() {
+		cloudProvider.InstanceTypes = fake.InstanceTypesAssorted()
+		nodePool = test.NodePool()
+		nodeClass = test.NodeClass(v1test1.TestNodeClass{
+			ObjectMeta: metav1.ObjectMeta{Name: nodePool.Spec.Template.Spec.NodeClassRef.Name},
+		})
+	})
+	AfterEach(func() { cloudProvider.NodeClassGroupVersionKind = nil })
+	It("should have status condition on nodePool as not ready due to error when getting nodeClass", func() {
+		ExpectApplied(ctx, env.Client, nodePool)
+		_ = ExpectObjectReconcileFailed(ctx, env.Client, nodePoolReadinessController, nodePool)
```

If the error is a NotFound error, should we return an error? In actuality, requeueing is going to be pointless and there isn't really an error here outside of the status condition. I could see us logging once, but I'm not sure it's worth throwing it back